sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -317,58 +323,221 @@ BlobClient blobClient = containerClient.GetBlobClient(blobName);
+```csharp
+CloudBlobContainer cloudBlobContainer; // properly configured reference to an existing container
+string blobPrefix; // prefix to list on
+IEnumerable<IListBlobItem> results = cloudBlobContainer.ListBlobs(prefix: blobPrefix);
+foreach (IListBlobItem item in results)
+{
+    // process blob listing
+}
+```
+
 v12
+
+v12 has explicit methods for listing by hierarchy.
 ```csharp
-// Get a reference to the container
-BlobContainerClient container = new BlobContainerClient(connectionString, containerName);
-// List all blobs in the container
-await foreach (BlobItem blobItem in containerClient.GetBlobsAsync())
+BlobContainerClient containerClient; // properly configured client to an existing container
+string blobPrefix; // prefix to list on
+IAsyncEnumerable<BlobHierarchyItem> results = containerClient.GetBlobsByHierarchyAsync(prefix: blobPrefix);
+await foreach (BlobHierarchyItem item in results)
 {
-    Console.WriteLine("\t" + blobItem.Name);
+    // process blob listing
 }
 ```
 
-### Other
+### Generate a SAS
+
+ There are various SAS tokens that may be generated. Visit our documentation pages to learn more: [Create a User Delegation SAS](https://docs.microsoft.com/azure/storage/blobs/storage-blob-user-delegation-sas-create-dotnet), [Create a Service SAS](https://docs.microsoft.com/azure/storage/blobs/storage-blob-service-sas-create-dotnet), or [Create an Account SAS](https://docs.microsoft.com/azure/storage/common/storage-account-sas-create-dotnet?toc=/azure/storage/blobs/toc.json).
+
+v11
+
+The following example is for creating a SAS to a single blob in the legacy library, but this pattern is applicable to container SAS and service SAS as well.
+
+```csharp
+// blob to generate a SAS for, must be authenticated with shared key
+CloudBlob blob;
+
+// Create a new access policy and define its constraints.
+SharedAccessBlobPolicy sasPolicy = new SharedAccessBlobPolicy()
+{
+    // SAS will be valid immetiately until 24 hours from now
+    SharedAccessExpiryTime = DateTime.UtcNow.AddHours(24),
+    Permissions = SharedAccessBlobPermissions.Read |
+        SharedAccessBlobPermissions.Write |
+        SharedAccessBlobPermissions.Create
+    // other optional parameters specified here
+};
+
+// Generate a SAS with the given policy, scoped to this blob
+string sasBlobToken = blob.GetSharedAccessSignature(sasPolicy);
+
+// optionally combine with URI for a full, self-authenticated URI to the blob
+return blob.Uri + sasBlobToken;
+```
+
+You could also make a SAS using a predefined policy stored on the service, instead of defining it in code.
+
+```csharp
+string policyId; // the id of the stored policy
+sasBlobToken = blob.GetSharedAccessSignature(null, policyName);
+```
+
+v12
+
+The modern SDK uses a builder pattern for constructing a SAS token. Clients are not involved in the process.
```

Comment:
we should add @amnguye 's work here - how to generate SAS using blobClient. it's going out pretty soon.

Reply:
I've spoken with her about adding this when it becomes available. That probably won't make it into this PR.